### PR TITLE
Add quantumflow

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by BastianZim** [comment]
@conda-forge/help-python

This would be ready for review, thanks!

File: recipes/quantumflow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    - sympy >=1.6
+    - typing_extensions
+  run_constrained:
+    - cirq >=0.8.0
+    - qiskit >=0.24.0
+    - pyquil >=2.28.0
+
+test:
+  imports:
+    - examples
```
Comment (dopplershift) [inline recipes/quantumflow/meta.yaml:44]: If this is passing, that means it's installing the examples as an `examples` package/module in the global package namespace, which is not good.

Comment (dopplershift) [reply recipes/quantumflow/meta.yaml:44]: Yeah, I think `setup.cfg` needs something like:
```ini
package_dir =
    = quantumflow
```

Comment (BastianZim) [reply recipes/quantumflow/meta.yaml:44]: When installing with pip it doesn't show up though:
```zsh
Package            Version
------------------ -------
cycler             0.11.0
decorator          4.4.2
fonttools          4.28.2
importlib-metadata 4.8.2
kiwisolver         1.3.2
matplotlib         3.5.0
mpmath             1.2.1
networkx           2.6.3
numpy              1.21.4
opt-einsum         3.3.0
packaging          21.3
Pillow             8.4.0
pip                21.3.1
pyparsing          3.0.6
python-dateutil    2.8.2
quantumflow        1.2.0
scipy              1.7.3
setuptools         59.4.0
setuptools-scm     6.3.2
six                1.16.0
sympy              1.9
tomli              1.2.2
wheel              0.37.0
zipp               3.6.0
```

Comment (BastianZim) [reply recipes/quantumflow/meta.yaml:44]: Scratch that, it shows up in the packages folder. I'll put in a PR upstream but it seems like this is not necessarily conda-forge's problem. Or should we hold off with merging until then?

Comment (dopplershift) [reply recipes/quantumflow/meta.yaml:44]: Or *maybe* removing `examples/__init__.py`? I'm really not clear why that exists.

Comment (BastianZim) [reply recipes/quantumflow/meta.yaml:44]: I will open an issue and see what upstream thinks, let's see.

Comment (dopplershift) [reply recipes/quantumflow/meta.yaml:44]: Well, we don't want our users to end up with an `examples` folder in `site-packages`, so we should do *something* on the conda-forge side.

Comment (BastianZim) [reply recipes/quantumflow/meta.yaml:44]: Ok, le me try something - one sec.

Comment (BastianZim) [reply recipes/quantumflow/meta.yaml:44]: Unfortunately, the package is still installed and, to be honest, not completely sure why. If you have a second, would you mind having a look at my PR here as I'm slightly stuck: https://github.com/gecrooks/quantumflow-dev/pull/87 Thanks!

Comment (dopplershift) [reply recipes/quantumflow/meta.yaml:44]: Maybe the work-around specified in [this issue](https://github.com/pypa/setuptools/issues/2688) could work?

Comment (BastianZim) [reply recipes/quantumflow/meta.yaml:44]: Also doesn't work. Hmm, this is really strange. I'll ask upstream.

**Comment by BastianZim** [comment]
Waiting for new release

File: recipes/quantumflow/meta.yaml (hunk 1 above)
Comment (BastianZim) [reply recipes/quantumflow/meta.yaml:42]: New release fixed it: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=424116&view=logs&jobId=6f142865-96c3-535c-b7ea-873d86b887bd&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823

**Review by dopplershift — APPROVED** [review]

**Comment by BastianZim** [comment]
Thanks!

**Pull request merged**